docs: Update desktop publishing article (#18405)
unoplatform/uno: doc/articles/uno-publishing-desktop.md

- Click **Save**
- Click **Publish**

## Publish Using The CLI

On Windows/macOS/Linux, open a terminal in your `csproj` folder and run:

```shell
dotnet publish -f net8.0-desktop
```

If you wish to do a self-contained publish, run the following instead:

```shell
dotnet publish -f net8.0-desktop -r {{RID}} -p:SelfContained=true
```

Where {{RID}} specifies the chosen OS and Architecture (e.g. win-x64). Again when targeting Windows, cross-publishing is not supported.

### macOS App Bundles

We now support generating .app bundles on macOS machines. From the CLI run:

```shell
dotnet publish -f net8.0-desktop -p:PackageFormat=app
```

You can also do a self-contained publish with:

```shell
dotnet publish -f net8.0-desktop -r {{RID}} -p:SelfContained=true -p:PackageFormat=app
```

Where {{RID}} is either `osx-x64` or `osx-arm64`.

> [!NOTE]
> Code signing is planned but not supported yet.

### Snap Packages

We support creating .snap packages on **Ubuntu 20.04** or later.

#### Requirements

The following must be installed and configured:

- snapd
- core20 on Ubuntu 20.04
- core22 on Ubuntu 22.04

> - core24 on Ubuntu 24.04

Reviewer: Let's explain how to do this, or point to docs that do so.

- multipass
- lxd
  - current user must be part of the `lxd` group
  - `lxd init --minimal` or similar should be run
- snapcraft

> Docker may interfere with Lxd causing network connectivity issues, for solutions see:
> 
> https://documentation.ubuntu.com/lxd/en/stable-5.0/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker

Reviewer: ```suggestion
> [!NOTE]
> Docker may interfere with Lxd causing network connectivity issues, [read this article for a solution](https://documentation.ubuntu.com/lxd/en/stable-5.0/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker).
```

#### Publishing A Snap

To publish a snap, run:

```shell
dotnet publish -f net8.0-desktop -r {{RID}} -p:SelfContained=true -p:PackageFormat=snap
```

> Where {{RID}} is either `linux-x64` or `linux-arm64`.

Reviewer: ```suggestion
Where `{{RID}}` is either `linux-x64` or `linux-arm64`.
```

We generate snap manifests in classic confinement mode and a .desktop file by default.

If you wish to customize your snap manifest, you will need to pass the following MSBuild properties:

- SnapManifest

> - DesktopFile

Reviewer: Should we point to docs on both file formats?

> The .desktop filename MUST conform to the Desktop File spec.

Reviewer: ```suggestion
The `.desktop` filename MUST conform to the Desktop File spec.
```

> If you wish, you can generate a default snap manifest and desktop file, then tweak them.

Reviewer: How ?

> [!NOTE]

> > .NET 9 publishing and cross-publishing are not supported yet. We will support .NET 9 publishing soon.

Reviewer: ```suggestion
> .NET 9 publishing and cross-publishing are not supported as of Uno 5.5.
```

uid: uno.publishing.desktop
---

# Publishing Your App For Desktop

## Preparing For Publish

> - [Profile your app with Visual Studio](https://learn.microsoft.com/en-us/visualstudio/profiling)

Reviewer 2: ```suggestion
- [Profile your app with Visual Studio](https://learn.microsoft.com/visualstudio/profiling)
```
